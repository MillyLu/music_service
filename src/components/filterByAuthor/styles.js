
import styled from "styled-components";

export const Dropdown = styled.div`
position: absolute;
width: 248px;
height: 305px;
background: ${props => props.theme === "light" ? "#FFFFFF" : "#313131"}; 
border: ${props => props.theme === "light" ? "2px solid #AD61FF" : 'none'}; 
border-radius: 12px;
`;

export const DropdownPerformer = styled(Dropdown)`
left: 443px;
top: 288px;
`;

export const DropdownGenre = styled(Dropdown)`
left: 741px;
top: 288px;
`;

export const DropdownList = styled.ul`
display: flex;
margin: 34px;
flex-direction: column;
align-items: flex-start;
width: 180px;
gap: 30px;  
overflow: scroll;        
width: 180px;
height: 237px;
&::-webkit-scrollbar{
    width: 2px;
}
&::-webkit-scrollbar-thumb {
    max-height: 20px;
    background-color: #FFFFFF;
}
`;

export const Li = styled.li`
list-style: none;
`;

export const DropdownItem = styled(Li)`
cursor: pointer;
margin: 0 -2px;
padding: 2px 2px;
color: ${props => props.theme === "light" ? "#000000" : "#FFFFFF"};
:hover{
    border-color: #AD61FF;
    color: #AD61FF;
    text-decoration: underline;
}
:active{
    border-color: #AD61FF;
    color: #AD61FF;
    cursor: pointer;
}
`;