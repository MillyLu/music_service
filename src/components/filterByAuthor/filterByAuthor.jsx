import { useState, useEffect, useContext } from 'react';
import * as Styled from './styles';
import { ThemeContext } from '../../ThemeProvider';

export function FilterButtonAuthor(props) {
    const {theme} = useContext(ThemeContext); 
 
    const [authors, setAuthors] = useState([]);
    const [chosenAuthor, setChosenAuthor] = useState('');
 
    useEffect(() => {
 const object = {};
 // eslint-disable-next-line no-return-assign
 const unicAuthors = props.data.filter(({author}) =>(!object[author] && (object[author] = 1)));
     setAuthors(unicAuthors)
    },[])
    
 let value;
 
    const handleAuthorChoose = (e) => {
         // eslint-disable-next-line prefer-destructuring
         value = e.target.childNodes[0].data;
         setChosenAuthor(value); 
         console.log(chosenAuthor);
         
    }
 
    useEffect(() => {
     props.getTracksByAuthor(chosenAuthor);
 }, [chosenAuthor])
    
 
    const authorsList = authors.map((item) => 
    <Styled.DropdownItem key={item.id} value={item.author} theme={theme} onClick={handleAuthorChoose}>{item.author}</Styled.DropdownItem>
    )
     return(
         <Styled.DropdownPerformer theme={theme}>
 
                 <Styled.DropdownList>
                     {authorsList}
                   
                 </Styled.DropdownList>     
         </Styled.DropdownPerformer> 
     )
 }