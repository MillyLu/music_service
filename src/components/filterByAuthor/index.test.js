/* eslint-disable no-undef */


import * as ShallowRenderer from 'react-shallow-renderer';
import { FilterButtonAuthor } from './filterByAuthor';




describe('<FilterButtonAuthor/>', () => {
        it('should render successfully and match snapshot', () => {

             const shallowRenderer = ShallowRenderer.createRenderer();
             const renderer = shallowRenderer;
             renderer.render(<FilterButtonAuthor />);
             const result = renderer.getRenderOutput();
             expect(result).toMatchSnapshot();  

          })
      });