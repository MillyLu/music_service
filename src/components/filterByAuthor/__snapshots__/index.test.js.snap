// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilterButtonAuthor/> should render successfully and match snapshot 1`] = `
<Styled(styled.div)
  theme={0}
>
  <styled.ul />
</Styled(styled.div)>
`;
